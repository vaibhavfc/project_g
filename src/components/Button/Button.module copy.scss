.root {
    position: absolute;
    min-width: 2rem;
    height: 3rem;
    /* Primary / burgundy-100 */
    background: #97144D;
    border-radius: 0.5rem;
    text-transform: none;

    &:disabled {
       // border: 1px solid red;
       background: #B4B4B4;
       cursor: not-allowed;
    }
    
}

.label {
    position: relative;
    width: 100%;
    white-space: nowrap;
    z-index: 2;
    font-family: Inter, sans-serif;
    font-weight: 400;
    font-style: normal;
    font-size: 0.875rem;
    line-height: 1.25rem;
    text-align: center;
    letter-spacing: 0.016rem;
    /* Primary / white-100 */
    color:white;

  }

  .outlined {
      &:hover {
        /* Primary / burgundy-100 */
          background: #97144D;
         // border-radius: 0.5rem;
      }
  }

  .sizeSmall {
      height: 2rem;
      font-size: 0.5rem;
  }

  .sizeLarge {
      width: 196px;
      height: 3.375rem;
      
  }

  .labelSmall {
    font-size: 0.75rem;
  }

  .labelLarge {
    font-size: 1rem;
  }

  .labelDisabledSecondary {
    color: #B4B4B4;
  }

  .outlinedSecondary {
    background: #F9F9F9;
    border: none;
    &:hover {
       background: #F1F1F1;
       border: none;
    }
    &:disabled {
      background: #F9F9F9;
      border: none;
   }
  }

  .textSecondary {
    color: #97144D;
  }

  .colorPrimary {
    color: #FFFFFF;
    font-size: 1.5rem;
    
  }

  .colorSecondary {
    color: #97144D;
    font-size: 1.5rem;
  }

  .labelContainer {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-content: center;
    align-items: center;
  }

  .iconButton {
    display: flex;
    //margin-left: 	0.5rem;
  }

  .iconButtonRoot {
    width:48px;
  }

  .iconSizeSmall {
    width: 32px;
  }

  .iconSizeLarge {
    width: 54px;
  }

  .flashButton {
    width: 320px;
    border-radius: 0px;
  }

  .labelPadding {
    padding-right: 16px;
  }

  .textButton {
    border: 1px solid red;
  }

  .labelPaddingSmall {
    padding-right: 8px;
  }

  .fontSizeSmall{
    font-size: 1rem;
  }

  .labelContainerPadding{
    padding-left: 11px;
    padding-right: 11px;
  }