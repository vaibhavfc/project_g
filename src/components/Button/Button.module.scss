@import '../../sass/theme/default';

.root {
  background: $button-bg;
  border-radius: 0.5rem;
  min-width: auto;
  height: 3rem;
  text-transform: none;
  box-shadow: none;
  
  &:hover {
    @include burgundy-highlight-1;
    box-shadow: none;
  }
  
  // &:focus{
  //   @include burgundy-highlight-2;
  //   box-shadow: none;
  // }
 
  // &:active {
  //   background-color: $button-ripple;
  //   box-shadow: none;
  // }

  &:disabled {
    color: white;
    background-color: $button-disabled;
  }
}

.secondary {
  background: $button-outlined__secondary;
  color: $button-bg;

  &:hover {
    color: $button-bg;
    background: $button-outlined__secondary--hover;
  }
  &:focus {
    color: $button-bg;
    background: $button-outlined__secondary--focus;
  }
  &:active {
    color: $button-ripple;
    background: $button-outlined__secondary--hover;
  }
  &:disabled {
    color: $button-disabled !important;
    background: $button-outlined__secondary !important;
  }
}

.label {
  @include cta-button-2;
  color: $white;
  padding-left: 1rem;
  padding-right: 1rem;
  z-index: 5;
}

.label_large {
  @include cta-button-1;
  color: $white;
  z-index: 5;
}

.label_medium {
  @include cta-button-2;
  color: $white;
  z-index: 5;
}

.label_small {
  @include cta-button-3;
  color: $white;
  z-index: 5;
}

.label_secondary_large {
  @include cta-button-1;
  color: $button-bg;
  z-index: 5;
  
  &:disabled {
    color: $button-disabled;
  }
}
.label_secondary_medium {
  @include cta-button-2;
  color: $button-bg;
  z-index: 5;
  
  &:disabled {
    color: $button-disabled;
  }
}
.label_secondary_small {
  @include cta-button-3;
  color: $button-bg;
  z-index: 5;
  
  &:disabled {
    color: $button-disabled;
  }
}

.label_icon_small {
  padding-left: 0.5625rem;
  padding-right: 0.5625rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  z-index: 5;
}

.label_icon_medium {
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  z-index: 5;
}

.label_icon_large {
  padding-left: 0.875rem;
  padding-right: 0.875rem;
  z-index: 5;
}

.label_icon_small_secondary {
  color: $button-bg !important; 
  padding-left: 0.5625rem;
  padding-right: 0.5625rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  z-index: 5;
}

.label_icon_medium_secondary {
  color: $button-bg !important; 
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  z-index: 5;
}

.label_icon_large_secondary {
  color: $button-bg !important;  
  padding-left: 0.875rem;
  padding-right: 0.875rem;
  z-index: 5;
}

.outlined {
  padding: 0;
  border: none;
  // padding-top: 0.75rem;
  // padding-bottom: 0.75rem;
  &:disabled {
    background: $button-disabled;
    cursor: not-allowed;
    border: none;
  }
  &:hover {
    background: $button-bg;
    border: none;
  }
  &:active {
    border: none;
  }
  &:focus {
    /* Primary / burgundy-100 */
    // background: #ED1164;
    // border-radius: 0.5rem;
  }
}

.sizeSmall {
  // min-width: 7.3125rem;
  height: 2rem;
}
.sizeLarge {
  // min-width: 12.25rem;
  height: 3.375rem;
}

.flushButton {
  min-width: 20rem;
  border-radius: 0px;
  display: flex;
  justify-content: center;
}

.endIcon {
  margin: 0;
}

.iconSizeSmall {
  margin: 0;
  // width: 2rem;
  
  & > :first-child {
    font-size: 1rem;
  }
}

.iconSizeMedium {
  // width: 3rem;

  & > :first-child {
    font-size: 1.5rem;
  }
}

.iconSizeLarge {
  // width: 3.375rem; 
  
  .iconSizeLarge > :first-child {
    font-size: 1.5rem;
  }
}

.onlyIconLarge {
  display: flex;
  justify-content: center;
  width: 0.25rem !important;
}
.onlyIconMedium {
  display: flex;
  justify-content: center;
  width: 0.5rem !important;
}

.onlyIconSmall {
  display: flex;
  justify-content: center;
  width: 0.375rem !important;
}

.label_disabled{
  color: $button-disabled;
}

.textBtn {
  @include text-button;
  color: $button-ripple;
  background: none;
  min-width: auto;
  height: min-content;
  padding: 0;

  &:hover {
    background: none;
  }
  
  &:focus {    
    background: none;
  }
  
  &:active {    
    background: none;
  }

  &:visited {
    background: none;
  }

  &:disabled {
    color: $button-disabled;
    background: none;
  }
}

.progress {
  height: 100%;
  max-height: 100%;
  max-width: 100%;
  border-radius: 0.5rem 0.5rem 0 0;
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
}

.progPrimary {
  background: $button-ripple;
}

.progSecondary {
  background: $button-ripple-secondary;
}