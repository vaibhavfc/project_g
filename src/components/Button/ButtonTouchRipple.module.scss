@import '../../sass/theme/default';

.root {
    z-index: 2;
    opacity: 1;
    color: $button-ripple; 
  }

.rippleVisible {
   // opacity: 0.5;
    //border: 10px solid red;
  }
.buttonRipple {
    // color: green; 
}