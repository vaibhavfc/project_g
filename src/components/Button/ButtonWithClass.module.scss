.root {
  background: pink;
}

.label {
  color: red;
  margin-left: 400px;
}