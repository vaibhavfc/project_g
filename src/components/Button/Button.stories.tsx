// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import ButtonGroup from '@material-ui/core/ButtonGroup';
import Button from '.';

import classes from './ButtonWithClass.module.scss';

import { ButtonProps } from './Button.type';

export default {
  title: 'Components/Button',
  component: Button,
  parameters: {
    controls: {
      // disabled: true,
    },
  },
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Primary button',
};

export const PrimaryWithIcon = Template.bind({});
PrimaryWithIcon.args = {
  children: 'Primary button',
  withIcon: true,
};

export const Secondory = Template.bind({});
Secondory.args = {
  color: 'secondary',
  children: 'Secondary button',
};

export const SecondaryWithIcon = Template.bind({});
SecondaryWithIcon.args = {
  children: 'Secondary button',
  color: 'secondary',
  withIcon: true,
};

export const IconButton = Template.bind({});
IconButton.args = {
  hasIconOnly: true,
};

export const FlushButton = Template.bind({});
FlushButton.args = {
  children: 'Flushed button',
  isFlushButton: true,
};

export const CustomButton = Template.bind({});
CustomButton.args = {
  children: 'Custom button',
  classes,
};

const TemplatePlayground: Story<ButtonProps> = (args) => (
  <ButtonGroup><Button {...args} /></ButtonGroup>
);

export const Playground = TemplatePlayground.bind({});
Playground.args = {
  children: 'TODO Build Playground',
};
