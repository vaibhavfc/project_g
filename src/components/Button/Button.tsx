import React, { FC, forwardRef, useMemo } from 'react';
import { Button as MuiButton } from '@material-ui/core';
import classnames from 'classnames';
import AddIcon from '@material-ui/icons/Add';
import { ButtonProps, ButtonStylingProps } from './Button.type';

import classes from './Button.module.scss';
import rippleClasses from './ButtonTouchRipple.module.scss';

import { mergeClassesObjects } from '../../helpers/styling/mergeClassesObjects';

const ButtonWithInnerRef: FC<ButtonProps> = ({
  innerRef = null,
  classes: overrideClasses = {},
  children,
  disabled = false,
  size = 'medium',
  color = 'primary',
  TouchRippleProps = {},
  withIcon = false,
  hasIconOnly = false,
  isFlushButton = false,
  doneCallback = () => undefined,
  ...props
}) => {
  const mergedClasses = useMemo(
    () => mergeClassesObjects<ButtonStylingProps>(classes, overrideClasses),
    [overrideClasses],
  );
  return (
    <MuiButton
      classes={{
        root: classnames(mergedClasses.root, { [classes.flashButton]: isFlushButton && color === 'primary' && !hasIconOnly && !withIcon }),
        label: classnames(mergedClasses.label, {
          [classes.label_secondary]: color === 'secondary', [classes.label_only_text]: !hasIconOnly, [classes.label_icon_medium]: hasIconOnly && size === 'medium', [classes.label_icon_large]: hasIconOnly && size === 'large', [classes.label_icon_small]: hasIconOnly && size === 'small',
        }),
        endIcon: classnames({ [classes.endIcon]: hasIconOnly }),
        sizeSmall: mergedClasses.sizeSmall,
        sizeLarge: mergedClasses.sizeLarge,
        outlined: classnames(mergedClasses.outlined),
        outlinedSecondary: mergedClasses.outlinedSecondary,
        iconSizeSmall: mergedClasses.iconSizeSmall,
        iconSizeMedium: mergedClasses.iconSizeMedium,
        iconSizeLarge: mergedClasses.iconSizeLarge,

      }}
      TouchRippleProps={{ classes: rippleClasses, ...TouchRippleProps }}
      variant="outlined"
      size={size}
      color={color}
      disabled={disabled}
      endIcon={(withIcon || hasIconOnly) && <AddIcon />}
    >
      { !hasIconOnly && children}
    </MuiButton>
  );
};

const Button = forwardRef<HTMLButtonElement, Omit<ButtonProps, 'innerRef'>>(
  (props, ref) => <ButtonWithInnerRef innerRef={ref} {...props} />,
);
export default Button;
