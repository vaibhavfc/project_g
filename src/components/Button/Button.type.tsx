import { Ref } from 'react';
import { ButtonProps as MuiButtonProps, ButtonClassKey } from '@material-ui/core/Button';

export interface ButtonStylingProps extends Partial<Record<ButtonClassKey, string>> {
  /**
   * override label(children) when loading is true
   */
  loadingLabel?: string;
  /**
   * override label(children) appled to done layer (even when not visible)
   */
  doneLabel?: string;
  /**
 * override label(children) when done is true
 */
  doneLabelVisible?: string;
  /**
   * override root(button itself) when done is true
   */
  doneRoot?: string;

}

export interface ButtonProps extends MuiButtonProps {
  /**
   * overriden BaseProps innerRef to HTMLButtonElement
   */
  innerRef: Ref<HTMLButtonElement>;
  /**
   * replaces children with CircularProgress
   */
  loading?: boolean;
  /**
   * replaces the button with the done transition
   */
  done?: boolean;
  /**
   * callback fired once the done transition is completed
   */
  withIcon?: boolean;

  hasIconOnly?: boolean;

  isFlushButton?: boolean;

  doneCallback?: () => void;
}
