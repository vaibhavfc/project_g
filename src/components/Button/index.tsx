import Button from './Button';
import { ButtonProps } from './Button.type';

export type { ButtonProps };

export default Button;
