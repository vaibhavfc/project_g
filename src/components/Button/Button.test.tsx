import { render } from '@testing-library/react';

import renderer from 'react-test-renderer';

import Button from './Button';

describe('test', () => {
  it('Renders Button Compoents', () => {
    render(<Button>test</Button>);
  });

  it('Matches Snapshot of Button', () => {
    const tree = renderer.create(<Button> TEST</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
