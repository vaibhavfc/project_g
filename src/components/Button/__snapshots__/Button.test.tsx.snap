// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Matches Snapshot of Button 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label label_only_text"
  >
     TEST
  </span>
</button>
`;
